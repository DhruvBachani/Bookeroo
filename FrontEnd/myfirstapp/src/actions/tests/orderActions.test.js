import React from 'react';
import reducer from '../../../src/reducers'
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();
const store = mockStore([]);

const getUserTransactions = () => ({
    type: 'GET_ERRORS'
});

const getAllTransactions = () => ({
    type: 'GET_ERRORS'
});

const refundOrder = () => ({
    type: 'GET_ERRORS'
});

const getShoppingCart = () => ({
    type: 'GET_ERRORS'
});

const getSellers = () => ({
    type: 'GET_ERRORS'
});

const saveOrder = () => ({
    type: 'GET_ERRORS'
});

describe('orderActions', () => {
    it('creates GET_ERRORS when user transactions cannot be retrieved', () => {
        store.dispatch(getUserTransactions());
        expect(store.getActions()).toMatchSnapshot();
    })

    it('creates GET_ERRORS when all transactions cannot be retrieved', () => {
        store.dispatch(getAllTransactions());
        expect(store.getActions()).toMatchSnapshot();
    })

    it('creates GET_ERRORS when a refund is unable to be executed', () => {
        store.dispatch(refundOrder());
        expect(store.getActions()).toMatchSnapshot();
    })

    it('creates GET_ERRORS when the shopping cart is unable to be retrieved', () => {
        store.dispatch(getShoppingCart());
        expect(store.getActions()).toMatchSnapshot();
    })

    it('creates GET_ERRORS when sellers are unable to be retrieved', () => {
        store.dispatch(getSellers());
        expect(store.getActions()).toMatchSnapshot();
    })

    it('creates GET_ERRORS when orders are unable to be saved', () => {
        store.dispatch(saveOrder());
        expect(store.getActions()).toMatchSnapshot();
    })
})