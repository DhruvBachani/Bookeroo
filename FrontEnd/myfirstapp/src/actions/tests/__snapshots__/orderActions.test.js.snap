// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`orderActions creates GET_ERRORS when a refund is unable to be executed 1`] = `
Array [
  Object {
    "type": "GET_ERRORS",
  },
  Object {
    "type": "GET_ERRORS",
  },
  Object {
    "type": "GET_ERRORS",
  },
]
`;

exports[`orderActions creates GET_ERRORS when all transactions cannot be retrieved 1`] = `
Array [
  Object {
    "type": "GET_ERRORS",
  },
  Object {
    "type": "GET_ERRORS",
  },
]
`;

exports[`orderActions creates GET_ERRORS when orders are unable to be saved 1`] = `
Array [
  Object {
    "type": "GET_ERRORS",
  },
  Object {
    "type": "GET_ERRORS",
  },
  Object {
    "type": "GET_ERRORS",
  },
  Object {
    "type": "GET_ERRORS",
  },
  Object {
    "type": "GET_ERRORS",
  },
  Object {
    "type": "GET_ERRORS",
  },
]
`;

exports[`orderActions creates GET_ERRORS when sellers are unable to be retrieved 1`] = `
Array [
  Object {
    "type": "GET_ERRORS",
  },
  Object {
    "type": "GET_ERRORS",
  },
  Object {
    "type": "GET_ERRORS",
  },
  Object {
    "type": "GET_ERRORS",
  },
  Object {
    "type": "GET_ERRORS",
  },
]
`;

exports[`orderActions creates GET_ERRORS when the shopping cart is unable to be retrieved 1`] = `
Array [
  Object {
    "type": "GET_ERRORS",
  },
  Object {
    "type": "GET_ERRORS",
  },
  Object {
    "type": "GET_ERRORS",
  },
  Object {
    "type": "GET_ERRORS",
  },
]
`;

exports[`orderActions creates GET_ERRORS when user transactions cannot be retrieved 1`] = `
Array [
  Object {
    "type": "GET_ERRORS",
  },
]
`;
